### ISSUE TITLE
[BUG] ios app does not sync photos to server

### ISSUE THREAD

**Issue by snomile** [immich-app/immich#2053]
### The bug

it seems there are some errors within the iOS app local database, new photos doesn't sync to server,  because the app believe all photos are already synced, except they are not.

pic1: a lot of pictures are out of sync
pic2: error message in iOS app
pic3: iOS app shows there are 0 picture need to sync

![A4F53091-00D0-4805-9B57-71B4EBB8F438](https://user-images.githubusercontent.com/4364681/227120163-06edcb4e-87c7-4fb3-8f8b-ca086b526658.png)

![4987145B-8BEA-4539-8462-842C15F02246](https://user-images.githubusercontent.com/4364681/227120180-43538de5-8e54-4cb6-88d7-aa00b40f0c26.png)

![268B2E39-A153-40BC-8FE1-C9EA64342F19](https://user-images.githubusercontent.com/4364681/227120200-f5a31a4a-59dd-4fcc-987e-aeb2fdda5f88.png)

### The OS that Immich Server is running on

Ubuntu 22.04

### Version of Immich Server

1.51.2

### Version of Immich Mobile App

1.51.0 build.89

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:release
    entrypoint: ["/bin/sh", "./start-server.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - redis
      - database
      - typesense
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:release
    entrypoint: ["/bin/sh", "./start-microservices.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - redis
      - database
      - typesense
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:release
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - model-cache:/cache
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    restart: always

  immich-web:
    container_name: immich_web
    image: ghcr.io/immich-app/immich-web:release
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    env_file:
      - .env
    restart: always

  typesense:
    container_name: immich_typesense
    image: typesense/typesense:0.24.0
    environment:
      - TYPESENSE_API_KEY=${TYPESENSE_API_KEY}
      - TYPESENSE_DATA_DIR=/data
    logging:
      driver: none
    volumes:
      - tsdata:/data
    restart: always

  redis:
    container_name: immich_redis
    image: redis:6.2
    restart: always

  database:
    container_name: immich_postgres
    image: postgres:14
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      PG_DATA: /var/lib/postgresql/data
    volumes:
      - /opt/immich/pgdata:/var/lib/postgresql/data
    #volumes:
    #  - pgdata:/var/lib/postgresql/data
    restart: always

  immich-proxy:
    container_name: immich_proxy
    image: ghcr.io/immich-app/immich-proxy:release
    environment:
      # Make sure these values get passed through from the env file
      - IMMICH_SERVER_URL
      - IMMICH_WEB_URL
    ports:
      - 2283:8080
    logging:
      driver: none
    depends_on:
      - immich-server
    restart: always

volumes:
  pgdata:
  model-cache:
  tsdata:
```

### Your .env content

```Shell
###################################################################################
# Database
###################################################################################

DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_PASSWORD=postgres
DB_DATABASE_NAME=immich

# Optional Database settings:
# DB_PORT=5432

###################################################################################
# Redis
###################################################################################

REDIS_HOSTNAME=immich_redis

# Optional Redis settings:
# REDIS_PORT=6379
# REDIS_DBINDEX=0
# REDIS_PASSWORD=
# REDIS_SOCKET=

###################################################################################
# Upload File Location
#
# This is the location where uploaded files are stored.
###################################################################################

UPLOAD_LOCATION=/media/media/Picture

###################################################################################
# JWT SECRET
#
# This JWT_SECRET is used to sign the authentication keys for user login
# You should set it to a long randomly generated value
# You can use this command to generate one: openssl rand -base64 128
###################################################################################

JWT_SECRET=xxxxxxxxxxxx

###################################################################################
# Typesense
###################################################################################
TYPESENSE_API_KEY=xxxxxxx
# TYPESENSE_ENABLED=false


###################################################################################
# Reverse Geocoding
#
# Reverse geocoding is done locally which has a small impact on memory usage
# This memory usage can be altered by changing the REVERSE_GEOCODING_PRECISION variable
# This ranges from 0-3 with 3 being the most precise
# 3 - Cities > 500 population: ~200MB RAM
# 2 - Cities > 1000 population: ~150MB RAM
# 1 - Cities > 5000 population: ~80MB RAM
# 0 - Cities > 15000 population: ~40MB RAM
####################################################################################

DISABLE_REVERSE_GEOCODING=false
REVERSE_GEOCODING_PRECISION=3

####################################################################################
# WEB - Optional
#
# Custom message on the login page, should be written in HTML form.
# For example:
# PUBLIC_LOGIN_PAGE_MESSAGE="This is a demo instance of Immich.<br><br>Email: <i>demo@demo.de</i><br>Password: <i>demo</i>"
####################################################################################

PUBLIC_LOGIN_PAGE_MESSAGE=

####################################################################################
# Alternative Service Addresses - Optional
#
# This is an advanced feature for users who may be running their immich services on different hosts.
# It will not change which address or port that services bind to within their containers, but it will change where other services look for their peers.
# Note: immich-microservices is bound to 3002, but no references are made
####################################################################################

IMMICH_WEB_URL=http://immich-web:3000
IMMICH_SERVER_URL=http://immich-server:3001
IMMICH_MACHINE_LEARNING_URL=http://immich-machine-learning:3003

####################################################################################
# Alternative API's External Address - Optional
#
# This is an advanced feature used to control the public server endpoint returned to clients during Well-known discovery.
# You should only use this if you want mobile apps to access the immich API over a custom URL. Do not include trailing slash.
# NOTE: At this time, the web app will not be affected by this setting and will continue to use the relative path: /api
# Examples: http://localhost:3001, http://immich-api.example.com, etc
####################################################################################

#IMMICH_API_URL_EXTERNAL=http://localhost:3001
```

### Reproduction steps

```bash
1.take a photo
2.hit start backup button
3.nothing happened except errors in log
```

**Comment by alextran1502** [comment]
@fyfrey do you know what can cause such issue with violate unique index in Isar?

**Comment by snomile** [comment]
PS: my mobile network was bad at the time, a lot of interruption durning uploading, so I manually stop and start the backup process for many times.

**Comment by fyfrey** [comment]
@alextran1502 there are three cases that could cause a unique key violation I can think of:

1. implementation bug
2. concurrent modification
3. there are multiple assets with the same combination of deviceId and localId (deviceAssetId)

@snomile can you please share the log with the error here? on the Logs screen there is a share button in the top right. it might contain more useful information


@snomile Have you checked whether the assets are backed up / uploaded to the server via Immich web? Maybe there are already backed up although its not visible in the app (due to the error present in the logs)

**Comment by snomile** [comment]
1. I tried to export the log via share button in the top right, but failed with message "Failed to load file".
2. I can't find the missing photos in immich web, so I am sure they are not synced.

**Comment by fyfrey** [comment]
Thank you for checking.

@alextran1502 The backup procedure is still completely independent from the local database (and its sync operations). So, the issue seems to also be in the backup code...
...or there are assets with duplicate IDs on the device: the backup service considers these as already uploaded and the sync service throws errors because of unique key violations. But iOS should never give out duplicate IDs for different assets - not sure what's the actual issue here.

**Comment by fyfrey** [comment]
I could possibly reproduce the error: The issue occurs if the server database contains multiple assets that have the same `deviceAssetId` and `deviceId`.

**Comment by PratikMahajan** [comment]
i saw a similar bug. 
**To reproduce,** 
1. take a photo
2. duplicate the photo in photos app
3. edit the duplicated photo. 

if 1 photo is already backed up and the 2nd photo is the last one in the backup queue, it'll get stuck at 100% and wont upload until you delete the other photo on immich. 

**Workaround:** 
1. download the photo from immich to files app or similar 
2. delete the photo on immich
3. upload the edited photo
4. add the downloaded photo from files app to photos 
5. upload the deleted photo.

**Comment by fyfrey** [comment]
@snomile does the Log screen now show any error messages regarding the failed Log export?

**Comment by snomile** [comment]
> @snomile does the Log screen now show any error messages regarding the failed Log export?

There're no logs about log export failure. 

additional info: If I share the log via AirDrop,  it shows 'fail' under device/user name. If I save the log to Files, nothing happend. If I save the log to Seafile Cloud client(a cloud storage service) , it shows 'Failed to load file'

**Comment by fyfrey** [comment]
this is *probably* the same issue as #2063

**Comment by alextran1502** [comment]
fixed in #2085
